clients/src/main/java/org/apache/kafka/common/protocol/types/Type.java

```diff
@@ -688,8 +688,10 @@ public Object read(ByteBuffer buffer) {
             if (size > buffer.remaining())
                 throw new SchemaException("Error reading bytes of size " + size + ", only " + buffer.remaining() + " bytes available");
 
+            int limit = buffer.limit();
+            buffer.limit(buffer.position() + size);
```

Comment:
nit: Since we will use `buffer.position() + size` twice, should we make it as a variable?

Reply:
I introduced a new variable `newPosition`